src/gui/userstatus.h

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (C) by Camila <hello@camila.codes>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License
+ * for more details.
+ */
+
+#ifndef USERSTATUS_H
+#define USERSTATUS_H
+
+#include <QPointer>
+#include "accountfwd.h"
+
+namespace OCC {
+
+class JsonApiJob;
+
+class UserStatus : public QObject
+{
+    Q_OBJECT
+    
+public:
+    explicit UserStatus(QObject *parent = nullptr);
+    enum Status {
+        Online,
+        DoNotDisturb,
+        Away,
+        Offline,
+        Invisible
+    };
+    Q_ENUM(Status);
+    void fetchUserStatus(AccountPtr account);
+    Status status() const;
+    QString message() const;
+    QUrl icon() const;
+
+private slots:
+    void slotFetchUserStatusFinished(const QJsonDocument &json, const int statusCode);
+
+signals:
+    void fetchUserStatusFinished();
+
+private:
+    Status stringToEnum(const QString &status) const;
+    
+    // it needs to match the Status enum
+    const QHash<QString, int> _preDefinedStatus{{"online", 0},
```

Comment:
I think I commented it already but I'd move this to a local variable in stringToEnum of type QHash<QString, Status>